src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -180,12 +233,24 @@ void rtc_set_alarm(datetime_t *t, rtc_callback_t user_callback) {
     irq_set_enabled(RTC_IRQ, true);
 
     rtc_enable_alarm();
+    return true;
 }
 
-void rtc_disable_alarm(void) {
-    // Disable matching and wait for it to stop being active
-    hw_clear_bits(&rtc_hw->irq_setup_0, RTC_IRQ_SETUP_0_MATCH_ENA_BITS);
-    while(rtc_hw->irq_setup_0 & RTC_IRQ_SETUP_0_MATCH_ACTIVE_BITS) {
-        tight_loop_contents();
+void rtc_delete_alarm(void)
+{
+    if (_callback == NULL) {
+        return; // rtc_set_alarm was not called or not successful
     }
```

Comment:
Feels like this function would be more robust if this check was done near the end rather than near the start? (or simply remove the check altogether, since `_callback` later gets set to NULL anyway)

Reply:
yes, I removed the user_callback check and therefore it must be removed here too. But this leads to the next issue, see
irq_remove_handler()/irq_set_exclusive_handler(). Seems like a NULL handler is handled as a valid function ptr there.